From 9bac71c21cb305449f3cce892c6e52d314137aa5 Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew_moseley@mentor.com>
Date: Sat, 5 Jul 2014 12:23:08 -0400
Subject: [PATCH 3/7] init-install.sh: Switch to using static device paths.

Use static device paths for mounting from /dev/disk/<blah> in
case the device ordering changes.

Signed-off-by: Drew Moseley <drew_moseley@mentor.com>
---
 meta/recipes-core/initrdscripts/files/init-install.sh | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

Upstream-Status: Submitted

diff --git a/meta/recipes-core/initrdscripts/files/init-install.sh b/meta/recipes-core/initrdscripts/files/init-install.sh
index 2cdac79..ec12f98 100644
--- init-install.sh
+++ init-install.sh
@@ -166,6 +166,12 @@ mkfs.ext3 $rootfs
 echo "Formatting swap partition...($swap)"
 mkswap $swap
 
+# Determine the static device paths
+sync; udevadm settle
+bootfs_static="$(udevadm info --query=property --name=$bootfs | grep DEVLINKS= | cut -d= -f2 | cut -d\  -f1)"
+swap_static="$(udevadm info --query=property --name=$swap | grep DEVLINKS= | cut -d= -f2 | cut -d\  -f1)"
+rootfs_uuid="$(udevadm info --query=property --name=$rootfs | grep ID_PART_ENTRY_UUID | cut -d= -f2)"
+
 mkdir /tgt_root
 mkdir /src_root
 mkdir -p /boot
@@ -176,8 +182,8 @@ mount -o rw,loop,noatime,nodiratime /run/media/$1/$2 /src_root
 echo "Copying rootfs files..."
 cp -a /src_root/* /tgt_root
 if [ -d /tgt_root/etc/ ] ; then
-    echo "$swap                swap             swap       defaults              0  0" >> /tgt_root/etc/fstab
-    echo "$bootfs              /boot            ext3       defaults              1  2" >> /tgt_root/etc/fstab
+    echo "$swap_static                swap             swap       defaults              0  0" >> /tgt_root/etc/fstab
+    echo "$bootfs_static              /boot            ext3       defaults              1  2" >> /tgt_root/etc/fstab
     # We dont want udev to mount our root device while we're booting...
     if [ -d /tgt_root/etc/udev/ ] ; then
 	echo "/dev/${device}" >> /tgt_root/etc/udev/mount.blacklist
@@ -196,7 +202,7 @@ if [ -f /etc/grub.d/00_header ] ; then
     cat >$GRUBCFG <<_EOF
 menuentry "Linux" {
     set root=(hd0,1)
-    linux /vmlinuz root=$rootfs $rootwait rw $5 $3 $4
+    linux /vmlinuz root=PARTUUID=$rootfs_uuid $rootwait rw $5 $3 $4
 }
 _EOF
     chmod 0444 $GRUBCFG
@@ -211,7 +217,7 @@ if [ ! -f /boot/grub/grub.cfg ] ; then
     echo "timeout 30" >> /boot/grub/menu.lst
     echo "title Live Boot/Install-Image" >> /boot/grub/menu.lst
     echo "root  (hd0,0)" >> /boot/grub/menu.lst
-    echo "kernel /vmlinuz root=$rootfs rw $3 $4" >> /boot/grub/menu.lst
+    echo "kernel /vmlinuz root=PARTUUID=$rootfs_uuid rw $3 $4" >> /boot/grub/menu.lst
 fi
 
 cp /run/media/$1/vmlinuz /boot/
-- 
1.9.1

